HPXMLtoOpenStudio/resources/schedules.rb

```diff
@@ -735,25 +775,109 @@ def self.set_weekend_rule(rule)
     rule.setApplySunday(true)
   end
 
-  def self.set_vacancy_periods(schedule, sch_name, vacancy_periods, year)
-    return if vacancy_periods.nil?
+  def self.set_off_periods(schedule, sch_name, off_periods, year)
+    return if off_periods.nil?
+
+    # Add off rule(s), will override previous rules
+    off_periods.each_with_index do |op, i|
+      value = 0.0
+
+      # Temperature of tank < 2C indicates of possibility of freeze.
+      value = 2.0 if sch_name.include?('WH Setpoint Temp') || sch_name.include?("#{Constants.ObjectNameWaterHeater} Setpoint")
+
+      next if sch_name.include?(Constants.ObjectNameNaturalVentilation) && op.natvent.nil?
+
+      value = 1.0 if sch_name.include?(Constants.ObjectNameNaturalVentilation) && op.natvent
+
+      day_s = Schedule.get_day_num_from_month_day(year, op.begin_month, op.begin_day)
+      day_e = Schedule.get_day_num_from_month_day(year, op.end_month, op.end_day)
+
+      op_begin_hour = op.begin_hour
+      op_end_hour = op.end_hour
+      if op_end_hour == 0
```

Comment:
Where did op_end_hour get defaulted to zero when it's not provided... did I miss it? Can it still be nil here? (I think the defaulting should occur in hpxml_defaults.rb.)